ext/qiita_marker/qfm_custom_block.c

```diff
@@ -120,21 +87,33 @@ static cmark_node *try_opening_qfm_custom_block_block(
 
     cmark_node *custom_block_node = cmark_parser_add_child(
         parser, parent_container, CMARK_NODE_QFM_CUSTOM_BLOCK,
-        parser->first_nonspace_column);
+        parser->first_nonspace + 1);
     custom_block_node->as.opaque = (node_qfm_custom_block *)parser->mem->calloc(
         1, sizeof(node_qfm_custom_block));
 
     cmark_strbuf *info = parser->mem->calloc(1, sizeof(cmark_strbuf));
-    cmark_strbuf_init(parser->mem, info, len - matched);
-    cmark_strbuf_put(info, input + matched, len - matched);
-    cmark_strbuf_trim(info);
 
-    set_qfm_custom_block_opening(custom_block_node, true);
+    /* Length from after opening custom block fence to before newline character.
+     */
+    bufsize_t info_len = len - matched;
+    if (info_len > 0 && input[len - 1] == '\n') {
+      info_len -= 1;
+    }
+    if (info_len > 0 && input[len - 1] == '\r') {
+      info_len -= 1;
+    }
+
+    cmark_strbuf_init(parser->mem, info, info_len);
+    cmark_strbuf_put(info, input + matched, info_len);
+    cmark_strbuf_trim(info);
     set_qfm_custom_block_info(custom_block_node, (char *)info->ptr);
 
     cmark_node_set_syntax_extension(custom_block_node, self);
     cmark_parser_advance_offset(parser, (char *)input,
-                                len - cmark_parser_get_offset(parser), 0);
+                                cmark_parser_get_first_nonspace(parser) +
+                                    matched + info_len -
+                                    cmark_parser_get_offset(parser),
+                                0);
```

Comment:
[memo]
`offset` was advanced to include the length of the newline character, which is why the `sourcepos` was incorrect.